tests/csapi/room_messages_test.go

```diff
@@ -59,6 +59,24 @@ func TestSendAndFetchMessage(t *testing.T) {
 	})
 }
 
+// With a non-existent room_id, GET /rooms/:room_id/messages returns 403
+// forbidden ("You aren't a member of the room").
+func TestFetchMessagesFromNonExistentRoom(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	roomID := "!does-not-exist:hs1"
+
+	// then request messages from the room
+	queryParams := url.Values{}
+	queryParams.Set("dir", "b")
+	res := alice.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "messages"}, client.WithQueries(queryParams))
+	must.MatchResponse(t, res, match.HTTPResponse{
+		StatusCode: http.StatusForbidden,
```

Comment:
Looks like Dendrite doesn't match the spec either at the moment. Should we skip for now?

> `room_messages_test.go:75: MatchResponse got status 500 want 403 - http://localhost:49256/_matrix/client/r0/rooms/%21does-not-exist:hs1/messages?dir=b => {"errcode":"M_UNKNOWN","error":"Internal Server Error"}`
>
> *-- https://github.com/matrix-org/complement/runs/6362103594?check_suite_focus=true#step:8:632*

Reply:
Wait a bit and we'll get a Dendrite fix in :)

Reply:
https://github.com/matrix-org/dendrite/pull/2450 🎉